Tests/DatadogTests/Datadog/Mocks/SystemFrameworks/FoundationMocks.swift

```diff
@@ -50,12 +50,16 @@ extension Data: AnyMockable, RandomMockable {
         return Data(repeating: byte, count: count)
     }
 
-    static func mock(ofSize size: UInt64) -> Data {
+    static func mock<Size>(ofSize size: Size) -> Data where Size: BinaryInteger {
         return mockRepeating(byte: 0x41, times: Int(size))
     }
 
+    static func mockRandom<Size>(ofSize size: Size) -> Data where Size: BinaryInteger {
+        return mockRepeating(byte: .random(in: 0x00...0xFF), times: Int(size))
```

Comment:
Definitely, good catch!

Reply:
@ncreated, I've been adding a better randomisation using [SecRandomCopyBytes](https://developer.apple.com/documentation/security/1399291-secrandomcopybytes) BUT all compression started failing.. So I've been reading more about the subject and discovered that random data (especially cryptographically strong random numbers) **cannot** be compressed. In [this wikipedia article](https://en.wikipedia.org/wiki/Lossless_compression#Lossless_compression_techniques) you can read: 
> The primary encoding algorithms used to produce bit sequences are Huffman coding (also used by the deflate algorithm) and arithmetic coding. Arithmetic coding achieves compression rates close to the best possible for a particular statistical model, which is given by the information entropy, whereas Huffman compression is simpler and faster but produces poor results for models that deal with symbol probabilities close to 1.

> In particular, files of random data cannot be consistently compressed by any conceivable lossless data compression algorithm; indeed, this result is used to define the concept of randomness in Kolmogorov complexity.[18]
> It is probably impossible to create an algorithm that can losslessly compress any data

In conclusion, a good randomised array of bytes won't be compressible.

Reply:
we can create such random strings by selecting `randomElement()`s from a pool of keywords
along these lines:
```swift
let keywords = ["foo", "bar", ...]
let randomString = (1...100).map { _ in keywords.randomElement()! } // this should be much longer than keywords.count
let compressed = zip(randomString)
// do assertions
```
that's also closer to the actual payloads where the same keys are usually repeated multiple times.

Reply:
>picking a random byte N times, instead of picking a random byte and repeating it N times.

Yes, but that's still fails most of the time. Instead, I've added an object to encode with random properties. This perform a good randomisation with enough redundancy to allow compression.